src/axom/slam/tests/slam_set_BivariateSet.cpp

```diff
@@ -0,0 +1,467 @@
+  {
+    return m_pset1;
+  }
+
+  // Template magic to access the appropriate second set based on
+  // the SecondSetType. When SecondSetType is slam::Set, we return PositionSet
+  template <typename S>
+  typename std::enable_if<std::is_abstract<S>::value, S*>::type getSecondSet()
+  {
+    return m_pset2;
+  }
+
+  template <typename S>
+  typename std::enable_if<std::is_same<S, RSet1>::value, S*>::type getSecondSet()
+  {
+    return m_rset2;
+  }
+
+  template <typename S>
+  typename std::enable_if<std::is_same<S, ISet1>::value, S*>::type getSecondSet()
+  {
+    return m_iset2;
+  }
+
+  template <typename S>
+  typename std::enable_if<std::is_same<S, PSet1>::value, S*>::type getSecondSet()
+  {
+    return m_pset2;
+  }
+
+protected:
+  FirstSetType* m_set1;
+  SecondSetType* m_set2;
+
+  PSet1* m_pset1;
+  PSet2* m_pset2;
+
+  RSet1* m_rset1;
+  RSet2* m_rset2;
+
+  ISet1* m_iset1;
+  ISet2* m_iset2;
+
+  Vec setIndices1;
+  Vec setIndices2;
+
+  Vec relationBegins;
+  Vec relationIndices;
+  RelationType modRelation;
+};
+
+// Tests several types of BivariateSet that differ in their underlying set types
+using MyTypes = ::testing::Types<
+  slam::BivariateSet<>,
+  slam::BivariateSet<slam::PositionSet<>, slam::PositionSet<>>,
+  slam::BivariateSet<slam::PositionSet<>, slam::RangeSet<>>,
+  slam::BivariateSet<slam::RangeSet<>, slam::PositionSet<>>,
+  slam::BivariateSet<slam::RangeSet<>, slam::RangeSet<>>,
+  slam::BivariateSet<slam::Set<>, slam::RangeSet<>>,
+  slam::BivariateSet<slam::PositionSet<>, slam::Set<>>,
+  slam::BivariateSet<slam::VectorIndirectionSet<>, slam::VectorIndirectionSet<>>,
+  slam::BivariateSet<slam::VectorIndirectionSet<>>,
+  slam::BivariateSet<slam::VectorIndirectionSet<>, slam::RangeSet<>>,
+  slam::BivariateSet<slam::RangeSet<>, slam::VectorIndirectionSet<>>>;
+TYPED_TEST_SUITE(BivariateSetTester, MyTypes);
+
+//-----------------------------------------------------------------------------
+
+TYPED_TEST(BivariateSetTester, smoke)
+{
+  using BSet = typename TestFixture::BivariateSetType;
+  using S1 = typename TestFixture::FirstSetType;
+  using S2 = typename TestFixture::SecondSetType;
+
+  // Test a default-constructed BivariateSet
+  {
+    BSet* bset = new slam::ProductSet<S1, S2>();
+
+    // We expect this object to be valid when the types of both S1 and S2 are
+    // abstract slam::Sets, but not if either is specialized.
+    bool isSet1Abstract = std::is_abstract<S1>::value;
+    bool isSet2Abstract = std::is_abstract<S2>::value;
+
+    EXPECT_EQ(isSet1Abstract && isSet2Abstract, bset->isValid(true));
+    delete bset;
+  }
+
+  // Test a non-default-constructed BivariateSet
+  {
+    BSet* bset = new slam::ProductSet<S1, S2>(this->m_set1, this->m_set2);
+    EXPECT_TRUE(bset->isValid(true));
+    delete bset;
+  }
+}
+
+//-----------------------------------------------------------------------------
+
+// Tests BivariateSeet::firstSetSize(), secondSetSize()
```

Comment:
```suggestion
// Tests BivariateSet::firstSetSize(), secondSetSize()
```
Check for copy/paste of this typo elsewhere